test/test_foreach.py

```diff
@@ -702,6 +702,15 @@ def test_foreach_l2_large_value_input(self, device, dtype, op):
             self.assertTrue(all(torch.isinf(e) for e in expect))
         self.assertEqual(expect, actual, equal_nan=False)
 
+    @onlyCUDA
+    @ops(foreach_reduce_op_db)
+    def test_foreach_reduce_large_input(self, device, dtype, op):
+        # test inputs larger than kChunkSize = 65536
+        ord, N = 2, 65536 * 2
+        n_expected_kernels = 3 if dtype in floating_types_and(torch.half, torch.bfloat16) else 1
```

Comment:
`3` because `foreach` implementation launches cleanup kernels? Also, probably doesn't make sense to test for all dtypes?

Reply:
3 is the expected number of kernels launched for the fast path, I copied it from the existing `test_reduce_fastpath`. I believe the norm is computed using multiple kernels.
I think it's fine to test on all dtypes, the test is quick, but let me know if I should switch to float32.

Reply:
```
pytest -vv --durations=6 -k test_foreach_reduce_large_input test/test_foreach.py


0.06s call     test/test_foreach.py::TestForeachCUDA::test_foreach_reduce_large_input__foreach_norm_cuda_bfloat16
0.00s call     test/test_foreach.py::TestForeachCUDA::test_foreach_reduce_large_input__foreach_norm_cuda_float16
0.00s call     test/test_foreach.py::TestForeachCUDA::test_foreach_reduce_large_input__foreach_norm_cuda_float32
0.00s call     test/test_foreach.py::TestForeachCUDA::test_foreach_reduce_large_input__foreach_norm_cuda_complex64
0.00s call     test/test_foreach.py::TestForeachCUDA::test_foreach_reduce_large_input__foreach_norm_cuda_float64
0.00s call     test/test_foreach.py::TestForeachCUDA::test_foreach_reduce_large_input__foreach_norm_cuda_complex128
```
And the first slow one is probably because of CUDA initialization